test/unit/ngsiv2/provisioning/device-registration_test.js

```diff
@@ -0,0 +1,344 @@
+ * If not, seehttp://www.gnu.org/licenses/.
+ *
+ * For those usages not covered by the GNU Affero General Public License
+ * please contact with::[contacto@tid.es]
+ *
+ * Modified by: Daniel Calvo - ATOS Research & Innovation
+ */
+'use strict';
+
+var iotAgentLib = require('../../../../lib/fiware-iotagent-lib'),
+    utils = require('../../../tools/utils'),
+    should = require('should'),
+    logger = require('logops'),
+    nock = require('nock'),
+    async = require('async'),
+    contextBrokerMock,
+    iotAgentConfig = {
+        contextBroker: {
+            host: '192.168.1.1',
+            port: '1026',
+            ngsiVersion: 'v2'
+        },
+        server: {
+            port: 4041
+        },
+        types: {
+            'Light': {
+                commands: [],
+                lazy: [
+                    {
+                        name: 'temperature',
+                        type: 'centigrades'
+                    }
+                ],
+                active: [
+                    {
+                        name: 'pressure',
+                        type: 'Hgmm'
+                    }
+                ],
+                service: 'smartGondor',
+                subservice: 'gardens'
+            },
+            'Termometer': {
+                commands: [],
+                lazy: [
+                    {
+                        name: 'temp',
+                        type: 'kelvin'
+                    }
+                ],
+                active: [
+                ],
+                service: 'smartGondor',
+                subservice: 'gardens'
+            }
+        },
+        service: 'smartGondor',
+        subservice: 'gardens',
+        providerUrl: 'http://smartGondor.com',
+        deviceRegistrationDuration: 'P1M',
+        throttling: 'PT5S'
+    },
+    device1 = {
+        id: 'light1',
+        type: 'Light',
+        service: 'smartGondor',
+        subservice: 'gardens'
+    },
+    device2 = {
+        id: 'term2',
+        type: 'Termometer',
+        service: 'smartGondor',
+        subservice: 'gardens'
+    };
+
+describe('IoT Agent Device Registration', function() {
+    beforeEach(function() {
+        logger.setLevel('FATAL');
+    });
+
+    afterEach(function(done) {
+        iotAgentLib.clearAll(function() {
+            // We need to remove the registrationId so that the library does not consider next operatios as updates.
+            delete device1.registrationId;
+            delete device2.registrationId;
+            iotAgentLib.deactivate(done);
+        });
+    });
+
+    describe('When a new device is connected to the IoT Agent', function() {
+        beforeEach(function(done) {
+            nock.cleanAll();
+
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', 'gardens')
+                .post('/v2/entities?options=upsert')
```

Comment:
Mock using post without payload. Comment explaining it needed?

Reply:
Fixed with fbc86fa76f4561b711a63053695efd21941cf4e5